app/src/pages/prompt/promptLoader.tsx

```diff
@@ -26,5 +26,7 @@ export async function promptLoader(args: LoaderFunctionArgs) {
     {
       id: promptId as string,
     }
-  ).toPromise();
+  )
+    .toPromise()
+    .catch(() => new Response("Prompt not found", { status: 404 }));
```

Comment:
This displays a human readable error message when you enter a bad slug. We should upgrade the ErrorComponent to display a Not Found page if the error is instance of Response and has status 404